pkg/middleware/org_redirect.go

```diff
@@ -44,12 +44,13 @@ func OrgRedirect(cfg *setting.Cfg) web.Handler {
 			return
 		}
 
-		var queryString string = c.Req.URL.Query().Encode()
-		if strings.Contains(queryString, "kiosk=") && !strings.Contains(queryString, "kiosk=tv") {
-			queryString = strings.Replace(queryString, "kiosk=", "kiosk", -1)
+		urlParams := c.Req.URL.Query()
+
+		if urlParams.Has("kiosk") && urlParams.Get("kiosk") == "" {
+			urlParams.Set("kiosk", "true")
```

Comment:
If you keep `kiosk` parameter it will still be encoded as `kiosk=`.
Instead you can do something like:
```suggestion
		qs := urlParams.Encode()
		if c.Req.URL.Query().Has("kiosk") {
			urlParams.Del("kiosk")
			qs = fmt.Sprintf("%s&kiosk", urlParams.Encode())
		}
```